From fca2d38b51ec8acbb93a58748d245793c1010913 Mon Sep 17 00:00:00 2001
From: Ivan Chavero <ichavero@redhat.com>
Date: Mon, 4 Aug 2014 13:46:21 -0600
Subject: [PATCH] Fixes plugin.ini error

When executing the cisco.pp manifest it throws an error
if the plugin.ini file is already declared.

I know this plugin will be deprecated in juno but this change
is needed for icehouse.

Change-Id: I9b8efe668911013d4fd801def15d0c4c384e42ca
---
 manifests/plugins/cisco.pp                 | 14 +++++++++-----
 spec/classes/neutron_plugins_cisco_spec.rb |  3 +--
 2 files changed, 10 insertions(+), 7 deletions(-)

diff --git a/manifests/plugins/cisco.pp b/manifests/plugins/cisco.pp
index 54ceda5..db19096 100644
--- a/manifests/plugins/cisco.pp
+++ b/manifests/plugins/cisco.pp
@@ -171,9 +171,13 @@ class neutron::plugins::cisco(
 
   # In RH, this link is used to start Neutron process but in Debian, it's used only
   # to manage database synchronization.
-  ensure_resource('file', '/etc/neutron/plugin.ini', {
-    ensure  => link,
-    target  => '/etc/neutron/plugins/openvswitch/ovs_neutron_plugin.ini',
-    require => Package['neutron-plugin-ovs']
-  })
+  if defined(File['/etc/neutron/plugin.ini']) {
+    File <| path == '/etc/neutron/plugin.ini' |> { target => '/etc/neutron/plugins/cisco/cisco_plugins.ini' }
+  }
+  else {
+    file {'/etc/neutron/plugin.ini':
+      ensure  => link,
+      target  => '/etc/neutron/plugins/cisco/cisco_plugins.ini',
+    }
+  }
 }
diff --git a/spec/classes/neutron_plugins_cisco_spec.rb b/spec/classes/neutron_plugins_cisco_spec.rb
index 2438342..d56e31f 100644
--- a/spec/classes/neutron_plugins_cisco_spec.rb
+++ b/spec/classes/neutron_plugins_cisco_spec.rb
@@ -42,8 +42,7 @@ describe 'neutron::plugins::cisco' do
     it 'should create plugin symbolic link' do
       should contain_file('/etc/neutron/plugin.ini').with(
         :ensure  => 'link',
-        :target  => '/etc/neutron/plugins/openvswitch/ovs_neutron_plugin.ini',
-        :require => 'Package[neutron-plugin-ovs]'
+        :target  => '/etc/neutron/plugins/cisco/cisco_plugins.ini'
       )
     end
 
-- 
1.9.3

